airbyte-cdk/python/unit_tests/sources/file_based/scenarios/csv_scenarios.py

```diff
@@ -180,11 +180,26 @@
                                                     "default": 0,
                                                     "type": "integer",
                                                 },
-                                                "autogenerate_column_names": {
-                                                    "title": "Autogenerate Column Names",
-                                                    "description": "Whether to autogenerate column names if column_names is empty. If true, column names will be of the form \u201cf0\u201d, \u201cf1\u201d\u2026 If false, column names will be read from the first CSV row after skip_rows_before_header.",
-                                                    "default": False,
-                                                    "type": "boolean",
+                                                "header_definition": {
```

Comment:
How can I validate this output? I'm especially suspicious about the:
```
"default": {
    "definition_type": "From CSV", "column_names": None
},
```

Reply:
While testing, it felt like the UX wasn't optimal as it would create two "Optional Fields" collapsed sections because the UI treat `type: object` differently than other types. 

<img width="920" alt="image" src="https://github.com/airbytehq/airbyte/assets/3360483/139ba60e-7079-4657-8c39-09dc8f4b5303">

To avoid confusion, we did move this back at the root of CsvFormat

Reply:
This is the new look without the two `Optional fields` 
<img width="485" alt="image" src="https://github.com/airbytehq/airbyte/assets/3360483/dd5ea473-cb0a-4ab9-8d36-0ae8250fca34">